lib/pure/math.nim

```diff
@@ -880,6 +880,33 @@ func floorMod*[T: SomeNumber](x, y: T): T =
   result = x mod y
   if (result > 0 and y < 0) or (result < 0 and y > 0): result += y
 
+func euclDiv*[T: SomeInteger](x, y: T): T {.since: (1, 5, 1).} =
+  ## Returns euclidean division of `x` by `y`.
+  runnableExamples:
+    assert euclDiv(13, 3) == 4
+    assert euclDiv(-13, 3) == -5
+    assert euclDiv(13, -3) == -4
+    assert euclDiv(-13, -3) == 5
+  result = x div y
+  let r = x mod y
+  if r < 0:
+    if y > 0:
+      dec result
+    else:
+      inc result
```

Comment:
```suggestion
  if x < 0:
    if y > 0:
      dec result
    else:
      inc result
```

Reply:
Why?

Reply:
It doesn't need the `x mod y` step; see https://en.wikipedia.org/wiki/Modulo_operation#/media/File:Divmod.svg

Reply:
> Note that any decent C compiler optimizes a division followed by a
> modulus into a single division/modulus instruction.

Reply:
Reference implementation
https://www.microsoft.com/en-us/research/wp-content/uploads/2016/02/divmodnote-letter.pdf
https://github.com/nothings/stb/blob/master/stb_divide.h

Reply:
And pre-existing `floorDiv` also use this way.
https://github.com/nim-lang/Nim/blob/6b69ff9eb64122e3e31269bf8ed35b64cf4cdc9c/lib/pure/math.nim#L864

Reply:
You could make a following PR to clean both of them. But I think these two ways are almost the same.

Reply:
Actually, disregard this; my version is correct for cases where x and y are negative and equal